src/Common/Temporal.WorkflowClient/WorkflowStubAttributes.cs

```diff
@@ -0,0 +1,124 @@
+    /// Parameters are not validated by the client and are sent to the workflow as provided.
+    /// If no parameters are provided, an empty payload is sent.
+    /// 
+    /// If a RunMethodStub is invoked on a stub instance that is NOT yet bound to a workflow run, it will attempt to bind the stub
+    /// instance to the first option permitted by both, the 'WorkflowRunStubConfiguration' specified to the stab creation method and
+    /// the settings of this attribute in the following order:
+    ///   1) Bind to an existing Active run
+    ///   2) Start a New run and bind to it
+    ///   3) Bind to an existing Finished run
+    /// If it cannot find anything to bind to based on above-mentioned permissions, an error is thrown.
+    /// If permissions specify a binding strategy, but the execution fails, the respective error is thrown
+    /// and no other binding is attempted. For example, assume that all CanBindXxx settings are True, and there are existing runs,
+    /// yet all of them are finished. In that case, based on the above order, it will try to start a New run and bind to it.
+    /// If, however, starting new run fails based on the 'WorkflowIdReusePolicy', the failure will be propagated and binding
+    /// to an existing Finished run will not be attempted.
+    /// 
+    /// If CanBindToExistingRun is true, then the RunMethodStub can be called to an active workflow run multiple times.
+    /// In that case the returned Task represents the completion of the active run; the run is not started again.
+    /// Potential arguments are ignored in such case.
+    /// 
+    /// If the 'MustBindToNewIfContinued' property of the 'WorkflowRunStubConfiguration' used when the stub instance was created is True,
+    /// then the Task returned by a RunMethodStub represents the completion of the ENTIRE workflow, including the as-new continuations
+    /// of the current run.
+    /// 
+    /// This and other 'WorkflowXxxStub' attributes can only be applied to method definitions in interfaces.
+    /// They are ignored in classes.
+    /// This and other 'WorkflowXxxStub' are interpreted by the workflow client SDK. They do NOT configure how
+    /// workflow implementations are hosted by the worker. However, if a worker host loads a workflow that implements interfaces
+    /// with 'WorkflowXxxStub'-attributed methods, the host will validate that the workflow implementation has attribute-based
+    /// handlers for APIs defined by the 'WorkflowXxxStub' attributes.
+    /// 
+    /// Note that the signatures required for a 'WorkflowXxxStubAttribute' may be different from the corresponding 'WorkflowXxxHandlerAttribute'.
+    /// For examples, queries are async from the client perspective, but must be handled synchronously in the implementation.
+    ///     
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, Inherited = true, AllowMultiple = true)]
+    public sealed class WorkflowRunMethodStubAttribute : Attribute
```

Comment:
Can you help me understand where this attribute is used? Also, I'd argue that we don't need the "stub" intermediary. I think top-level `client.ExecuteWorkflow<MyWorkflow>(myOptions)` and all the others are just fine.

Reply:
The stub generator needs to know which of the iface methods corresponds to the main workflow method. Or did I misunderstand the question?

Reply:
Using Java as the base, we've iterated over this API in TS for a long time and in my opinion it's the most polished.
Given that TS has a superior type system we might not be able to implement everything here but the fundamental concepts still hold.

- Start a workflow and get a handle: `WorkflowHandle<MyWorkflow> handle = await client.Start<MyWorkflow>(options)`
- Start a workflow and wait for the result: `MyResultType res = await client.Execute<MyWorkflow>(options)`
- Use a handle to signal the workflow: `var res = await handle.Signal<MySignal>(arg1, arg2)`
- handle also supports `terminate()`, `cancel(reason)`, getting the `result()`, and `query`
- Getting a handle for a running workflow: `var handle = client.GetHandle<MyWorkflow>(workflowId, runId /* optional */)`